File: new/2864_maximum-odd-binary-number.go
package main

import "strings"

func maximumOddBinaryNumber(s string) string {
	num := 0
	for _, c := range s {
		if c == '1' {
			num += 1
		}
	}
	if num == 0 || num == len(s) {
		return s
	}
	if num == 1 {
		return strings.Repeat("0", len(s)-1) + "1"
	}
	return strings.Repeat("1", num-1) + strings.Repeat("0", len(s)-num) + "1"
}
func main() {
	s := maximumOddBinaryNumber("0101")
	println(s)
}

File: rust/src/bin/2864_maximum-odd-binary-number.rs
#![allow(unused_variables, dead_code)]
struct Solution {}
impl Solution {
    pub fn maximum_odd_binary_number(s: String) -> String {
        let mut num = 0;
        for c in s.chars() {
            if c == '1' {
                num += 1;
            }
        }
        if num == 0 || s.len() == num {
            return s;
        }
        let mut res = "1".repeat(num - 1);
        res.push_str(&"0".repeat(s.len() - num));
        res.push_str(&"1");

        return res;
    }
}
fn main() {}
